src/engine/audio/audio.main.ts

```diff
@@ -419,10 +419,10 @@ function processAudioPlaybackRingBuffer(ctx: IMainThreadContext, audioModule: Ma
     audioPlaybackQueue.push(audioItem);
   }
 
-  for (let i = 0; i < audioPlaybackQueue.length; i++) {
+  for (let i = audioPlaybackQueue.length - 1; i >= 0; i--) {
```

Comment:
I think the previous method of iteration was working right? Not sure why we would need to change this. Do we want to iterate from the beginning or end?

Reply:
yes it was working, but i changed this during my investigation and it seemed a bit cleaner than having to decrement `i--` after the splice.

Reply:
Right, but don't we want to iterate in the same order that we pushed onto the stack? This is now a LIFO instead of a FIFO queue. The order these events are applied determines the audio state, so I'm not sure we can use a LIFO.

Reply:
i was gonna say that's why i changed `audioPlaybackQueue.push` to `audioPlaybackQueue.unshift` but forgot to push lol